### PR TITLE
Fix determination of staked QUIC connections

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
The usage of `ConnectionPeerType` to determine if a connection is staked is incorrect. The `peer_type` is an attribute of the `connection_table` where a QUIC connection gets stored. We maintain two tables, one for staked connections, and the other for unstaked connections. But if the staked connection is full, the new connections get stored in the unstaked connection table, even if the peer is a staked client. So `ConnectionPeerType` can incorrectly mark a staked connection as unstaked.

Peer's stake correctly identifies if the connection is from a staked client or not. 

#### Summary of Changes
Update code to use peer's stake to determine if the connection is from a staked client.
Remove `ConnectionPeerType` and its usage.

Fixes #

### CODE REVIEW

File: streamer/src/nonblocking/quic.rs [hunk 1]
```diff
@@ -720,19 +692,17 @@ async fn handle_connection(
     stats.total_connections.fetch_add(1, Ordering::Relaxed);
     let mut max_streams_per_throttling_interval =
         stream_throttle::max_streams_for_connection_in_throttling_duration(
-            peer_type,
             params.stake,
             params.total_stake,
             stream_load_ema.clone(),
         );
-    let staked_stream = matches!(peer_type, ConnectionPeerType::Staked) && params.stake > 0;
```
Comment (lijunwangs) [inline streamer/src/nonblocking/quic.rs:728]: We can reduce some of this change by defining:

let staked_stream = params.stake > 0;

Comment (pgarg66) [reply streamer/src/nonblocking/quic.rs:728]: Isn't it better to have this check than declaring a new stack variable? The change is only three lines. So review should be simple.

**Review by t-nelson — COMMENTED** [review]
i think i'd have preferred resolving the enum once and using that throughout. the `Staked` variant just needs to be `Staked { lamports: u64 }`

**Comment by pgarg66** [comment]
> i think i'd have preferred resolving the enum once and using that throughout. the `Staked` variant just needs to be `Staked { lamports: u64 }`

Yes, that's more idiomatic. Will update the PR.

**Comment by pgarg66** [comment]
> i think i'd have preferred resolving the enum once and using that throughout. the `Staked` variant just needs to be `Staked { lamports: u64 }`

Done

File: streamer/src/nonblocking/quic.rs [hunk 2]
```diff
@@ -83,6 +83,18 @@ struct PacketAccumulator {
...
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum ConnectionPeerType {
+    Unstaked,
+    Staked(u64),
+}
+
+impl ConnectionPeerType {
+    fn staked(&self) -> bool {
```
Comment (t-nelson) [inline streamer/src/nonblocking/quic.rs:93]:
```suggestion
    fn is_staked(&self) -> bool {
```

**Review by lijunwangs — APPROVED** [review]
lgtm

**Pull request merged**